src/model/users/rating_history.rs

```diff
@@ -0,0 +1,186 @@
+use serde::{
+    ser::{SerializeSeq, SerializeTuple},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+#[derive(Default, Clone, Debug, Serialize)]
+pub struct GetQuery {
+    pub trophies: bool,
+}
+
+pub type GetRequest = crate::model::Request<GetQuery>;
+
+impl GetRequest {
+    pub fn new(username: &str) -> Self {
+        Self::get(format!("/api/user/{username}/rating-history"), None, None)
+    }
+}
+
+impl<S: AsRef<str>> From<S> for GetRequest {
+    fn from(s: S) -> Self {
+        Self::new(s.as_ref())
+    }
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub enum RatingType {
+    Bullet,
+    Blitz,
+    Rapid,
+    Classical,
+    Correspondence,
+    Chess960,
+    #[serde(rename = "King of the Hill")]
+    KingOfTheHill,
+    #[serde(rename = "Three-check")]
+    ThreeCheck,
+    Antichess,
+    Atomic,
+    Horde,
+    #[serde(rename = "Racing Kings")]
+    RacingKings,
+    Crazyhouse,
+    UltraBullet,
+    Puzzles,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct RatingEntry {
+    pub name: RatingType,
+    pub points: Vec<RatingPoint>,
+}
+
+// Lichess returns a hard to use tuple of `[year, month, day, rating]`,
+// because of this Serialize / Deserialize are manually implemented.
+#[derive(Clone, Debug)]
```

Comment:
```suggestion
#[derive(Clone, Debug, Deserialize, Serialize)]
```
I've tested the above and it seems to work. Is there a reason not to do this?

Reply:
Deriving `Deserialize` seems fine (this is unexpected to me), but the tests won't pass since it won't serialize to an array by default, and thus a manual `Serialize` is necessary for correctness. There seems to be a crate for it [`serde_tuple`](https://crates.io/crates/serde_tuple), though taking a dependency for 12 lines of trivial code doesn't seem worth it.

Reply:
Fair, I forgot to check `Deserialize`